src/params/enum.js

```diff
@@ -0,0 +1,21 @@
+const assert = require('assert');
+const Str = require('./str');
+
+class Enum extends Str {
+  constructor(name, position, enums, opts = {}) {
+    super(name, position, opts);
+    assert(enums !== undefined, 'enums are required');
```

Comment:
Proper sentence structure please.
```suggestion
    assert(enums !== undefined, 'Enums are required.');
```